### PR TITLE
Refactor(eos_designs)!: Rename wan_mode 'autovpn' to 'legacy-autovpn'

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

The `wan_mode: autovpn` valid value is renamed `legacy-autovpn` in order to reuse the `autovpn` value for a later design.

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

* Rename the valid value in schema
* Adjust mention to AutoVPN in schema
* adjust code where it makes sense (renamed one function)
* update `how-to/wan.md`
* adjust molecule tests to reflect device names / filenames using `legacy-autovpn`
* fix some wrong documentation in eos_cli_config_gen schema
* updated release notes and porting guide

## How to test

molecule files changes are only related to the name of the devices. Everything else is passing

**Note:** no test asserting that `autovpn` is not supported was added. The error will show that valid values for `wan_mode` don't contain `autovpn` anymore.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 1]
```diff
@@ -315,6 +315,24 @@ The description can be reverted manually if needed:
 +mgmt_interface_description: "oob_management"
 ```
 
+#### `wan_mode: autovpn` renamed `wan_mode: legacy-autovpn`
+
+With AVD version 5.0.0 the valid values for `wan_mode` key have changed. If using the `autovpn` mode in AVD 4.x, the `wan_mode` key needs tp be updated to `legacy-autovpn`.
+
+```diff
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:322]: It is not interesting for a user to see the schema snip. They need to see what how their vars needs to change.

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 2]
```diff
@@ -315,6 +315,24 @@ The description can be reverted manually if needed:
 +mgmt_interface_description: "oob_management"
 ```
 
+#### `wan_mode: autovpn` renamed `wan_mode: legacy-autovpn`
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
### `wan_mode: autovpn` renamed `wan_mode: legacy-autovpn`
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 3]
```diff
@@ -315,6 +315,24 @@ The description can be reverted manually if needed:
...
+     documentation_options:
+       table: wan-settings
+-    description: Select if the WAN should be run using CV Pathfinder or AutoVPN only.
++    description: Select if the WAN should be run using CV Pathfinder or Legcay AutoVPN only.
+     type: str
+     default: cv-pathfinder
+     valid_values:
+-      - autovpn
++      - legacy-autovpn
+       - cv-pathfinder
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
-  wan_mode: autovpn
+  wan_mode: legacy-autovpn
```

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 4]
```diff
@@ -315,6 +315,15 @@ The description can be reverted manually if needed:
 +mgmt_interface_description: "oob_management"
 ```
 
+### `wan_mode: autovpn` renamed `wan_mode: legacy-autovpn`
+
+With AVD version 5.0.0 the valid values for `wan_mode` key have changed. If using the `autovpn` mode in AVD 4.x, the `wan_mode` key needs tp be updated to `legacy-autovpn`.
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
With AVD version 5.0.0 the valid values for `wan_mode` key have changed. If using the `autovpn` mode in AVD 4.x, the `wan_mode` key needs to be updated to `legacy-autovpn`.
```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/wan_mode.schema.yml [hunk 5]
```diff
@@ -9,9 +9,9 @@ keys:
   wan_mode:
     documentation_options:
       table: wan-settings
-    description: Select if the WAN should be run using CV Pathfinder or AutoVPN only.
+    description: Select if the WAN should be run using CV Pathfinder or Legcay AutoVPN only.
```
Comment (carlbuchmann) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/wan_mode.schema.yml]:
```suggestion
    description: Select if the WAN should be run using CV Pathfinder or Legacy AutoVPN only.
```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/wan_virtual_topologies.schema.yml [hunk 6]
```diff
@@ -84,7 +84,7 @@ keys:
         description: |-
           List of virtual toplogies policies.
 
-          For AutoVPN, each item in the list creates:
+          For Legcay AutoVPN, each item in the list creates:
```
Comment (carlbuchmann) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/wan_virtual_topologies.schema.yml]:
```suggestion
          For Legacy AutoVPN, each item in the list creates:
```

File: python-avd/pyavd/_eos_designs/structured_config/network_services/router_path_selection.py [hunk 7]
```diff
@@ -32,13 +32,13 @@ def router_path_selection(self: AvdStructuredConfigNetworkServices) -> dict | No
         }
 
         # When running CV Pathfinder, only load balance policies are configured
-        # for AutoVPN, need also vrfs and policies.
-        if self.shared_utils.wan_mode == "autovpn":
+        # for Legcay AutoVPN, need also vrfs and policies.
```
Comment (carlbuchmann) [inline python-avd/pyavd/_eos_designs/structured_config/network_services/router_path_selection.py]:
```suggestion
        # for Legacy AutoVPN, need also vrfs and policies.
```

File: python-avd/pyavd/_eos_designs/structured_config/network_services/router_path_selection.py [hunk 8]
```diff
@@ -68,8 +68,8 @@ def _wan_load_balance_policies(self: AvdStructuredConfigNetworkServices) -> list
 
         return load_balance_policies
 
-    def _autovpn_policies(self: AvdStructuredConfigNetworkServices) -> list:
-        """Return a list of policies for AutoVPN."""
+    def _legacy_autovpn_policies(self: AvdStructuredConfigNetworkServices) -> list:
+        """Return a list of policies for Legcay AutoVPN."""
```
Comment (carlbuchmann) [inline python-avd/pyavd/_eos_designs/structured_config/network_services/router_path_selection.py]:
```suggestion
        """Return a list of policies for Legacy AutoVPN."""
```

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**